src/run/runner/wall_time/perf/perf_executable.rs

```diff
@@ -62,3 +62,39 @@ pub fn get_working_perf_executable() -> Option<OsString> {
     debug!("perf is installed but not functioning correctly");
     None
 }
+
+/// Detects if the required perf events are available on this system.
+/// Returns Some("-e {cycles,cache-references,cache-misses}") if all three events are available,
+/// None otherwise.
+pub fn get_event_flags(perf_executable: &OsString) -> anyhow::Result<Option<String>> {
+    const CYCLES_EVENT_NAME: &str = "cycles";
+    const CACHE_REFERENCES_EVENT_NAME: &str = "cache-references";
+    const CACHE_MISSES_EVENT_NAME: &str = "cache-misses";
+    let output = Command::new(perf_executable)
+        .arg("list")
+        .output()
+        .context("Failed to run perf list")?;
+
+    let stdout = String::from_utf8_lossy(&output.stdout);
+
+    // Check for the three required events
+    let has_cycles = stdout.lines().any(|line| line.contains(CYCLES_EVENT_NAME));
```

Comment:
This check can easily return true, even if we don't support cycles because it's only a substring match. For example: 
```
  ref-cycles                                         [Hardware event]
```
would also return `true` despite not being the same event (not sure if it's that likely or if there are any systems where it's not available).

Maybe do it like this:
```suggestion
    let has_cycles = stdout.lines().any(|line| line.trim().starts_with(CYCLES_EVENT_NAME));
```

Reply:
Much better indeed!